src/state.py

```diff
@@ -327,3 +330,35 @@ def get_nginx_config(charm: ops.CharmBase) -> NginxConfig:
 
         configurations[config.hostname][config.path] = config
     return configurations
+
+
+def get_hostnames(charm: ops.CharmBase) -> tuple[Hostname, ...]:
```

Comment:
Any particular reason for returning a tuple here? Given we're generating this as a list it is probably safe to return it

Reply:
Since the collection of hostname does not change, I thought tuple was a better fit here.

Reply:
In this case we're building a list in the function so it is pretty safe just to return it